src/server/lib/__tests__/headers.test.ts

```diff
@@ -0,0 +1,71 @@
+import Redis from 'ioredis-mock';
+import request from 'supertest';
+import { defaultEnv } from './sharedConfig';
+
+// Override the default 5s max timeout for these tests because Supertest takes some time to run.
+jest.setTimeout(10000);
+
+describe('Content Security Policy headers', () => {
+  const loggerInfoMock = jest.fn();
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.resetModules();
+
+    jest.mock('@/server/lib/getAssets', () => ({
+      getAssets: () => ({
+        main: { js: 'mocked' },
+        vendors: { js: 'mocked' },
+        runtime: { js: 'mocked' },
+      }),
+    }));
+
+    jest.mock('@/server/lib/serverSideLogger', () => ({
+      logger: {
+        info: loggerInfoMock,
+      },
+    }));
+
+    jest.mock('@/server/lib/redis/redisClient', () => new Redis());
+  });
+  const getServerInstance = async () => {
+    // eslint-disable-next-line functional/immutable-data
+    process.env = defaultEnv;
+
+    // Start the application server.
+    const { default: server } = await import('@/server/index');
+
+    // Tests call this method on success and error to tear down the server.
+    const teardownServer = async () => {
+      // Remove the scheduled teardown because we're doing it now.
+      await new Promise((r) => server.close(r));
+    };
+
+    return { server, teardownServer };
+  };
+
+  it('should ', async () => {
+    const { server, teardownServer } = await getServerInstance();
+    // Check our base routes that have user input forms
+    ['/register', '/sign-in', '/reset-password'].forEach(async (route) => {
+      try {
+        const response = await request(server).get(route);
+        // Is the CSP header set?
+        expect(response.header).toHaveProperty('content-security-policy');
```

Comment:
I'm guessing this is just the first pass of testing as it's only checking these three routes. Do you plan to extend this base file further to cover all our routes?

Reply:
I was trying to work out the best way to do this - at the moment, these are the three core routes which have forms and where we definitely want our CSP to work, and most of the other routes we serve will redirect to /signin if the user doesn't log in. I could write the test in Cypress instead and do this more in-depth but as an initial pass I think this does the job - would love your feedback!